File: internal/content/testdata/bench/go/nc-happy-number.go
func is_happy(n int) bool {
	seen := map[int]bool{}
	for n != 1 {
		if seen[n] {
			return false
		}
		seen[n] = true
		total := 0
		for n > 0 {
			d := n % 10
			total += d * d
			n /= 10
		}
		n = total
	}
	return true
}

File: internal/content/testdata/bench/java/nc-happy-number.java
import java.util.*;

class Solution {
    public boolean is_happy(long n) {
        Set<Long> seen = new HashSet<>();
        while (n != 1) {
            if (seen.contains(n)) return false;
            seen.add(n);
            long total = 0;
            while (n > 0) {
                long digit = n % 10;
                total += digit * digit;
                n /= 10;
            }
            n = total;
        }
        return true;
    }
}
